### PR TITLE
Add notifications to schema

### LINKED ISSUE #1124: Create notifications in amplify schema
All notifications need to be sorted via date and each notification consists of:

- From (which Colony the notification comes from)
- Time
- Read/Unread flag
- Text

The text can contain mentions to users and actions and these should be linked

https://docs.amplify.aws/guides/api-graphql/query-with-sorting/q/platform/js/

### PR DESCRIPTION
**Pull request by thomassdk**
## Description

This PR adds the notification types and models to the amplify schema including the notification settings.

These models will undoubtably change as the feature develops but this PR gives a solid baseline to work from.

Resolves #1124

### CODE REVIEW

File: amplify/backend/api/colonycdapp/schema.graphql [hunk 1]
```diff
@@ -998,11 +998,91 @@ type Profile @model {
...
+
+"""
+An object holding the personal notification settings of a user
+"""
+type NotificationSettings {
+  """
+  Device registration tokens for push notifications
+  If this is null or the array is empty the user is not subscribed to push notifications
+  """
+  notificationTokens: [String]
```
Comment (jakubcolony) [inline amplify/backend/api/colonycdapp/schema.graphql:1023]: Can elements of this array be `null`?

File: amplify/backend/api/colonycdapp/schema.graphql [hunk 2]
```diff
@@ -998,11 +998,91 @@ type Profile @model {
...
+Notifications that a user may receive from a Colony
+"""
+type Notification @model {
+  """
+  User ID the Notification was sent to
+  """
+  userId: ID!
+    @index(
+      name: "byUser"
+      queryField: "notificationsByDate"
```
Comment (jakubcolony) [inline amplify/backend/api/colonycdapp/schema.graphql:1048]: `getNotificationsByDate` for consistency

File: amplify/backend/api/colonycdapp/schema.graphql [hunk 3]
```diff
@@ -998,11 +998,91 @@ type Profile @model {
...
+  """
+  The text of the notification with indications which string portions are links
+  eg. 'This signifies a {link}'
+  """
+  text: String!
+  """
+  The links connected to what is signified in the text
+  The order matches what is in the string
+  """
+  links: [String]
```
Comment (jakubcolony) [inline amplify/backend/api/colonycdapp/schema.graphql:1085]: Same as above, can the elements be `null`?

File: amplify/backend/api/colonycdapp/schema.graphql [hunk 4]
```diff
@@ -998,11 +998,91 @@ type Profile @model {
...
   user: User! @belongsTo(fields: ["id"])
+  """
+  List of a user's notifications
+  """
+  notifications: [Notification] @hasMany
+  """
+  Notification settings
+  """
+  notificationSettings: NotificationSettings
+}
```
Comment (jakubcolony) [inline amplify/backend/api/colonycdapp/schema.graphql:1013]: What's the reasoning behind putting notifications on the profile model while the notification model itself has a relationship with the user model?

**Review by jakubcolony — COMMENTED** [review]
Left a few comments

File: amplify/backend/api/colonycdapp/schema.graphql (hunk 4 above)
Comment (thomassdk) [reply amplify/backend/api/colonycdapp/schema.graphql:1013]: None this is an oversight on my part which was left over from a previous experimentation where I had attached the Notifications to the user model.

Over time as I worked on this feature I decided that the `Profile` model was the correct place for Notifications as it uses elements of that model (such as the user's email) and generally seems to be where we store information about the user. Happy to be argued against however :)

I've updated the model to change the notification model to have a relationship with the *profile* now.

**Pull request merged**